pkg/typeutil/comparison.go

```diff
@@ -38,3 +38,16 @@ func MinDuration(a, b time.Duration) time.Duration {
 	}
 	return b
 }
+
+// StringsEqual checks if two string slices are equal. Empyt slice and nil are considered equal.
+func StringsEqual(a, b []string) bool {
```

Comment:
Do we need the unit test?

Reply:
Too simple to test. The code was written by AI, which I feel is highly credible.